src/devicecontroller/DeviceController.ts

```diff
@@ -86,15 +86,15 @@ export default interface DeviceController {
 
   /**
    * Selects a video input device to use. The constraint may be a device id,
-   * `MediaTrackConstraint`, `MediaStream` (containing video track), or `null` to
-   * indicate no device. The promise will resolve indicating success or it will
-   * throw an appropriate error indicating the failure.
+   * `MediaTrackConstraint`, `MediaStream` (containing video track). It may also be an {@link VideoTransformDevice}
+   * to apply video transform.
+   * The promise will resolve indicating success or it will throw an appropriate error indicating the failure.
    */
   startVideoInput(device: VideoInputDevice): Promise<MediaStream | undefined>;
```

Comment:
Suggest removing `null` from `AudioInputDevice` type and add it in the `startAudioInput` API:
```ts
// Now
type AudioInputDevice = Device | AudioTransformDevice | null;
startAudioInput(device: AudioInputDevice): Promise<MediaStream | undefined>;

// Suggest
type AudioInputDevice = Device | AudioTransformDevice;
startAudioInput(device: AudioInputDevice | null): Promise<MediaStream | undefined>;
```

* Keep `AudioInputDevice` type and `VideoInputDevice` type consistent.
* Easier to tell the type difference between parameter of `startAudioInput` and `startVideoInput`.

```ts
type VideoInputDevice = Device | VideoTransformDevice;
startVideoInput(device: VideoInputDevice): Promise<MediaStream | undefined>;
```

Reply:
Feel like null is a valid input for audio so I would rather keep it that way. Audio and video work different so there is no need for exact consistency. In the future, where we can truly join with no audio then we can just drop it from AudioInputDevice.

Reply:
Although `null` is a valid input but in fact there is no `null` device, it's just a special case. The essence of `Device` and `TransformDevice` is `MediaStream` obtained from browser. And `null` isn't a valid way to request it. 

If the type is called `startAudioInputParamType`, then I think it's ok to add `null`. But now it stands for a type of device, so it makes more sense to me to remove it from `AudioInputDevice` type.

Reply:
Seem like niche change to me. Just like MediaTrackConstraints is not a real device. The end result is that null generates an audio input change so it is an input type. Device is kinda a bad name since it can be stream and other stuff that is not device. Maybe the name should be `AudioInputType` instead of `AudioInputDevice`  but I am not going to change it now.

Reply:
Agree, `device` is a bad naming. Yeah we can change this later.